### PR TITLE
GUI fixes

### PR DESCRIPTION
**Pull request by NikoOinonen**
A couple of small fixes for issues mentioned in #42 
- Default save path for output images and data is in the same directory as the input file.
- Force field viewer can save the force field to a file.
- Added a .desktop file for GUI for better integration with the desktop environment.
- Typing values does not update the plot before pressing return.
- General code clean-up.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,11 +78,17 @@ Additionally an OpenCL Installable Client Driver (ICD) for your compute device i
 
 Run the GUI application:
 ```sh
-./ppm-gui
+./GUI/ppm-gui
+```
+
+In order to make the GUI application appear in the system application menu and 'Open with' context menus, link the `ppm-gui` application to a location that is on PATH, e.g. `~/.local/bin`, and install the provided .desktop file. This can be achived by running the following in the repository root:
+```bash
+ln -s `realpath ./GUI/ppm-gui` $HOME/.local/bin
```
Comment (ondrejkrejci) [inline README.md:86]: I guess, that will change, once we will have the full pip way, but OK for now.

File: ppafm/GUIWidgets.py [hunk 2]
```diff
@@ -413,22 +419,27 @@ def __init__(self, parent=None, title="View Forcefield", width=5, height=4, dpi=
...
         sl.setCurrentIndex(sl.findText(components[2])); sl.currentIndexChanged.connect(self.updateView)
         sl.setToolTip(tooltips[0]); hbl.addWidget(sl)
-        hbl.addStretch(1)
 
+        hbr = QtWidgets.QHBoxLayout(); hb.addLayout(hbr)
         lb = QtWidgets.QLabel('z index:'); lb.setToolTip(tooltips[1]); hbr.addWidget(lb)
         bx = QtWidgets.QSpinBox(); bx.setRange(0, 200); bx.setValue(0)
         bx.valueChanged.connect(self.updateView); bx.setToolTip(tooltips[1])
         hbr.addWidget(bx); self.bxInd = bx
+        hbr.setContentsMargins(10, 0, 10, 0)
```
Comment (ondrejkrejci) [inline ppafm/GUIWidgets.py:434]: Why such a chages in Margins?

File: ppafm/GUIWidgets.py [hunk 3]
```diff
@@ -459,3 +470,35 @@ def updateView(self):
 
         if self.verbose > 0: print('FFViewer.updateView', ic, iz, data.shape)
         if self.verbose > 1: print('updateView time [s]', time.perf_counter() - t0)
+
+    def saveFF(self):
```
Comment (ondrejkrejci) [inline ppafm/GUIWidgets.py:474]: This is great, and very usefull!

File: ppafm/GUIWidgets.py [hunk 4]
```diff
@@ -413,22 +419,27 @@ def __init__(self, parent=None, title="View Forcefield", width=5, height=4, dpi=
...
         hb = QtWidgets.QHBoxLayout(); self.centralLayout.addLayout(hb)
-        hbl = QtWidgets.QHBoxLayout(); hb.addLayout(hbl)
-        hbr = QtWidgets.QHBoxLayout(); hb.addLayout(hbr)
 
+        hbl = QtWidgets.QHBoxLayout(); hb.addLayout(hbl)
         lb = QtWidgets.QLabel('Component:'); lb.setToolTip(tooltips[0]); hbl.addWidget(lb)
         sl = QtWidgets.QComboBox(); self.slComponent = sl; sl.addItems(components)
         sl.setCurrentIndex(sl.findText(components[2])); sl.currentIndexChanged.connect(self.updateView)
         sl.setToolTip(tooltips[0]); hbl.addWidget(sl)
-        hbl.addStretch(1)
```
Comment (ondrejkrejci) [inline ppafm/GUIWidgets.py:423]: I hope that you know, what you are doing here.

File: GUI/ppm-gui [hunk 5]
```diff
@@ -823,13 +641,293 @@ class ApplicationWindow(QtWidgets.QMainWindow):
...
+
+        # Left-right divide for labels and input boxes
+        vb = QtWidgets.QHBoxLayout(); self.l0.addLayout(vb)
+        bxl = QtWidgets.QVBoxLayout(); vb.addLayout(bxl, 1)
+        bxr = QtWidgets.QVBoxLayout(); vb.addLayout(bxr, 3)
+
+        # Lattice vector A
+        lb = QtWidgets.QLabel("A (x, y, z) [Å]"); lb.setToolTip(TTips['PBC']); bxl.addWidget(lb)
+        vb = QtWidgets.QHBoxLayout(); bxr.addLayout(vb)
+        self.bxPBCAx = _spin_box((-100, 100), 50, 0.1, self.updatePBC, TTips['PBC'], vb)
```
Comment (ondrejkrejci) [inline GUI/ppm-gui:787]: Let's hope that no one will have bigger systems than 100angstrom in near future!

**Review by ondrejkrejci — APPROVED** [review]
Just few super-small comments inside, otherwise I do not find any other problems.

File: README.md (hunk 1 above)
Comment (NikoOinonen) [reply README.md:86]: Yes, this is meant to be temporary.

File: ppafm/GUIWidgets.py (hunk 4 above)
Comment (NikoOinonen) [reply ppafm/GUIWidgets.py:423]: This and the following margin change were just to make it look a bit better with the addition of the button for saving the force field.

File: ppafm/GUIWidgets.py (hunk 2 above)
Comment (NikoOinonen) [reply ppafm/GUIWidgets.py:434]: See the other comment.

File: GUI/ppm-gui (hunk 5 above)
Comment (NikoOinonen) [reply GUI/ppm-gui:787]: Well, it's probably pretty rare to go above 100 Å, but I could just increase it up to 1000 Å.

Comment (ondrejkrejci) [reply GUI/ppm-gui:787]: I do not think so for now, on most of nowadays machines, such a big FF would lead to _memory overflow_ ...

Comment (NikoOinonen) [reply GUI/ppm-gui:787]: There could be cases where the periodic cell is big in one direction, but narrow in another, like some ribbons of 2D materials. Also, the lattice vector size here does not affect the memory usage. I think pyqt just requires the maximum values to determine the minimum size of the spin box. So there should be no harm in increasing the maximum value.

**Pull request merged**